### DISCUSSION TITLE
[Feature] Verify that charts are pulled from trusted sources - verify with gpg signed helm charts.

### DISCUSSION THREAD

**Discussion by dominikkaminski** [helmfile/helmfile#980] [Ideas]
Hey helmfile community and maintainers,

at first, thanks for this great open-source project - it helps us a lot!

For a project focused on security and verifiable software supply chains, we are looking for way to verify that helm charts are published from the vendor/community we trust. For example, we sign our to OCI registry published helm charts with the native `helm sign` method.

During native installation, we set `helm --verify --keyring <trusted keyring file>` arguments, so only trusted artifacts get installed.

We miss the same possibilities for the helmfile `repositories`. In my opinion, there should be a way to verify (`oci: true`) packages during pull and thus I miss a `verify` and `keyring` argument:
```
repositories:
  - name: "dont-trust-verify"
    oci: true
    url: "my.registry.local"
    verify: true
    keyring: "~/.gnupg/pubring.gpg"
```

**TL;DR;**
Before installing releases into a cluster, helm charts should be verifiable against gpg public key.

Thanks for taking this idea into consideration! :pray:

**Comment by yxxhero** [comment]
@dominikkaminski that is a good idea. PR is welcome.

**Comment by yxxhero** [comment]
@dominikkaminski please use the main branch. and it will be nice if you can give me some feedback.

**Comment by dominikkaminski** [comment]
@yxxhero Thank you very much for your extremely fast implementation of this feature request!

I directly tested it in the main branch - it works :partying_face: 

```
Pulling <chart>
Pulled: <chart>
Digest: sha256:fd6791dba49ac5811ac2750107ade6161020b5e9171028a75b32fd9621d1e31c
Pulled: <chart>
Digest: sha256:fd6791dba49ac5811ac2750107ade6161020b5e9171028a75b32fd9621d1e31c
Signed by: sign@<email>
Using Key With Fingerprint: 778D3EB0813A523936030025C3A9C3E9A87C8C44
Chart Hash Verified: sha256:61aeea992c16d7339d4a801d50f5938d1a0174eda8114b42b9338e2fc55bd65a
```

**Reply by yxxhero** [reply]
@dominikkaminski thanks. new release in 3 weeks.
